gammapy/cube/psf_kernel.py

```diff
@@ -0,0 +1,310 @@
+            the maximum radius of the PSF kernel. If float, we assume unit is degree.
+        normalize : bool
+            normalize the PSF kernel (per energy)
+        factor : int
+            the oversample factor to compute the PSF
+
+        Returns
+        -------
+        kernel : `~gammapy.cube.PSFKernel`
+            the kernel Map with reduced geometry according to the max_radius
+        """
+        if max_radius is not None:
+            # Create a new geom accordingly
+            center = geom.center_coord[:2]
+            binsz = Quantity(np.abs(geom.wcs.wcs.cdelt[0]), 'deg')
+            max_radius = Quantity(max_radius, 'deg')
+            npix = 2 * int(max_radius / binsz) + 1
+            geom = WcsGeom.create(skydir=center, binsz=binsz, npix=npix, proj=geom.projection,
+                                  coordsys=geom.coordsys, axes=geom.axes)
+        return cls(table_psf_to_kernel_map(table_psf, geom, normalize, factor))
+
+    @classmethod
+    def from_gauss(cls, geom, sigma, max_radius=None, containment_fraction=0.99,
+                   normalize=True, factor=4):
+        """Create Gaussian PSF.
+
+        This is used for testing and examples.
+        The map geometry parameters (pixel size, energy bins) are taken from `geom`.
+        The Gaussian width ``sigma`` is a scalar,
+
+        TODO : support array input if it should vary along the energy axis.
+
+        Parameters
+        ----------
+        geom : `~gammapy.map.WcsGeom`
+            Map geometry
+        sigma : `~astropy.coordinates.Angle` or float
+            Gaussian width. Assume degrees if float input
+        max_radius : `~astropy.coordinates.Angle` or float
+            Desired kernel map size. Implicit unit is degree.
+        normalize : bool
+            normalize the PSF kernel (per energy)
+        factor : int
+            the oversample factor to compute the PSF
+
+        Returns
+        -------
+        kernel : `~gammapy.cube.PSFKernel`
+            the kernel Map with reduced geometry according to the max_radius
+        """
+        sigma = Angle(sigma, 'deg')
+
+        if max_radius is None:
+            max_radius = Gauss2DPDF(sigma.to('deg').value).containment_radius(
+                containment_fraction=containment_fraction)
+
+        max_radius = Angle(max_radius, 'deg')
+
+        # Create a new geom according to given input
+        center = geom.center_coord[:2]
+        binsz = Quantity(np.abs(geom.wcs.wcs.cdelt[0]), 'deg')
+        npix = 2 * int(max_radius / binsz) + 1
+        geom = WcsGeom.create(skydir=center, binsz=binsz, npix=npix, proj=geom.projection,
+                              coordsys=geom.coordsys, axes=geom.axes)
+
+        rad = Angle(np.linspace(0., max_radius, 200), 'deg')
+
+        table_psf = TablePSF.from_shape(shape='gauss', width=sigma, rad=rad)
+
+        return cls(table_psf_to_kernel_map(table_psf, geom))
+
+    def to_map(self):
```

Comment:
Suggest to remove `to_map`. If we want to expose the `_psf_kernel_map`, we should rename it to `psf_kernel_map`, but not add a method with a different name that just returns this data member.

Reply:
Done